sdk/synapse/synapse-access-control/src/accessControlClient.ts

```diff
@@ -86,12 +86,9 @@ export class AccessControlClient extends AccessControlClientContext {
   private async _getRoleDefinitions(
     options?: coreHttp.OperationOptions
   ): Promise<AccessControlClientGetRoleDefinitionsResponse> {
-    const { span, updatedOptions } = createSpan(
-      "AccessControlClient-_getRoleDefinitions",
-      coreHttp.operationOptionsToRequestOptionsBase(options || {})
-    );
+    const { span, updatedOptions } = createSpan("AccessControlClient-_getRoleDefinitions", options);
```

Comment:
@joheredi - this is a decent sample of what I changed. There were two "convert operation options to request options base" methods - `coreHttp2.operationOptionsToRequestOptionsBase` (which this example is) and `this.getOperationOptions`.

In both cases the fix is the same:

1. Change createSpan to take the original `options` from the call (type OperationOptions)
2. Take the `updatedOptions` from `createSpan` and pass them to whichever conversion function was being used before. 
3. Assign that avlue to the `options` for the coreHttp.OperationArguments

It was tedious, but it's honest work. No innovation in any of these areas, just rote transformation.

Reply:
Well this is a great catch! Thanks Richard, I have filed Azure/autorest.typescript/issues/875 to fix this in the generator